### PR TITLE
[ie/SmotretTV] Add extractor

### LINKED ISSUE #8899: smotret.tv URLs
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

all

### Example URLs

https://smotret.tv/rossiya-1

### Provide a description that is worded well enough to be understood

does not work

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
reiner@reiner:~$ yt-dlp -vU https://smotret.tv/rossiya-1
[debug] Command-line config: ['-vU', 'https://smotret.tv/rossiya-1']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.11.16 from yt-dlp/yt-dlp [24f827875] (zip)
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-6.2.0-39-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, pyxattr-0.7.2, requests-2.25.1, secretstorage-3.3.1, sqlite3-3.37.2, urllib3-1.26.5
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1901 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/latest/download/_update_spec
[debug] Downloading SHA2-256SUMS from https://github.com/yt-dlp/yt-dlp/releases/download/2023.12.30/SHA2-256SUMS
Current version: stable@2023.11.16 from yt-dlp/yt-dlp
Latest version: stable@2023.12.30 from yt-dlp/yt-dlp
Current Build Hash: 0b122b45079b60ef720ebf4f72cbe53139f284b11492a42673cf6aa52d1432d8
Updating to stable@2023.12.30 from yt-dlp/yt-dlp ...
ERROR: Unable to write to /usr/local/bin/yt-dlp; try running as administrator
[generic] Extracting URL: https://smotret.tv/rossiya-1
[generic] rossiya-1: Downloading webpage
ERROR: [generic] Unable to download webpage: HTTP Error 520: None (caused by <HTTPError 520: None>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/generic.py", line 2439, in _real_extract
    full_response = self._request_webpage(url, video_id, headers=filter_dict({
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 860, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/usr/local/bin/yt-dlp/yt_dlp/networking/_urllib.py", line 396, in _send
    res = opener.open(urllib_req, timeout=float(request.extensions.get('timeout') or self.timeout))
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 520: 

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 4052, in urlopen
    return self._request_director.send(req)
  File "/usr/local/bin/yt-dlp/yt_dlp/networking/common.py", line 114, in send
    response = handler.send(request)
  File "/usr/local/bin/yt-dlp/yt_dlp/networking/_helper.py", line 204, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/networking/common.py", line 325, in send
    return self._send(request)
  File "/usr/local/bin/yt-dlp/yt_dlp/networking/_urllib.py", line 401, in _send
    raise HTTPError(UrllibResponseAdapter(e.fp), redirect_loop='redirect error' in str(e)) from e
yt_dlp.networking.exceptions.HTTPError: HTTP Error 520: None

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 847, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 4074, in urlopen
    raise _CompatHTTPError(e) from e
yt_dlp.networking.exceptions._CompatHTTPError: HTTP Error 520: None
```

### PR DESCRIPTION
**Pull request by pzhlkj6612**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixes #8899

New extractors for Russian TV websites.

| URL | Logo
|:- | -
| https://smotret.tv/ | Logo of smotret.tv
| http://smotret-tv.live/ | Logo of smotret-tv.live

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/smotrettv.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            'subtitles': dict,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        # RutubeEmbed in iframe
+        'url': 'https://smotret.tv/otr',
+        'info_dict': {
+            'id': 'faa934385b83f9e8a92f5484defae5fa',  # TODO: use "otr"
```
Comment (pzhlkj6612) [inline yt_dlp/extractor/smotrettv.py:37]: TODO: About `self.url_result()`, how do I preserve the id?

**Comment by seproDev** [comment]
Closing due to being an unofficial site

File: yt_dlp/extractor/rutube.py [hunk 2]
```diff
@@ -80,6 +81,8 @@ def _extract_formats(self, options, video_id):
                     'url': format_url,
                     'format_id': format_id,
                 })
+        for hls_url in traverse_obj(options, ('live_streams', 'hls', ..., 'url', {url_or_none})):
+            formats += self._extract_m3u8_formats(hls_url, video_id, ext='mp4')
```
Comment (pzhlkj6612) [inline yt_dlp/extractor/rutube.py:85]: TODO: move this change into another PR and add a test.

File: yt_dlp/extractor/smotrettv.py (hunk 1 above)
Comment (pzhlkj6612) [reply yt_dlp/extractor/smotrettv.py:37]: No way for now, see https://github.com/yt-dlp/yt-dlp/pull/10831#discussion_r1729607085.

File: yt_dlp/extractor/smotrettv.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+        logo_url = None
+        if postopis := get_element_by_class('postopis', webpage):
+            if logo_path := self._search_regex(
+                    r'<img\s+src\s*=\s*"(?P<path>[^"]+)".+?/>', postopis, 'logo path', group='path', default=None):
+                logo_url = urllib.parse.urljoin(base, logo_path)
+
+        player = self._download_webpage(f'http://cdntvmedia.com/{video_id}.php', video_id, headers={'Referer': base})
+        m3u8_urls = self._search_regex(
+            r'Playerjs\s*\(\s*{.+?\s*file:\s*(["\'])(?P<url>https://[^"\']+)\1', player, 'player', group='url')
```
Comment (pzhlkj6612) [inline yt_dlp/extractor/smotrettv.py:128]: Or:

```suggestion
        m3u8_urls = self._search_regex(
            r'Playerjs\s*\(\s*{.+?\s*file:\s*(["\'])(?P<url>https://(?:(?!\1).)+)\1', player, 'player', group='url')
```

To myself: do not forget what I learned.
